public/app/features/alerting/unified/components/rule-editor/DashboardPicker.tsx

```diff
@@ -52,6 +55,34 @@ interface DashboardPickerProps {
   onDismiss: () => void;
 }
 
+const useFilteredDashboards = (dashboardFilter: string) => {
+  const [isDashSearchFetching, setIsDashSearchFetching] = useState(false);
+  const [dashboards, setDashboards] = useState<DashboardQueryResult[]>([]);
+  const [locationInfo, setLocationInfo] = useState<Record<string, LocationInfo>>({});
+
+  useEffect(() => {
```

Comment:
I think we can simplify this with a `useAsync` to not have to set all the state vars.